front/gatsby/src/components/Layout/Header/Header.tsx

```diff
@@ -2,85 +2,71 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { useSubscription } from '@apollo/react-hooks';
-import { AccountsContext } from '@substrate/context';
+import { useQuery, useSubscription } from '@apollo/react-hooks';
+import { formatBalance } from '@polkadot/util';
+import { AccountsContext, ApiContext } from '@substrate/context';
 import { Button, ItemStats } from '@substrate/design-system';
-import gql from 'graphql-tag';
 import React, { useContext, useEffect, useState } from 'react';
 
 import { APP_TITLE, toShortAddress } from '../../../util';
+import {
+  BLOCKS_SUBSCRIPTION,
+  ERAS_SUBSCRIPTION,
+  LATEST_ERA_QUERY,
+  LATEST_SESSION_QUERY,
+  SESSIONS_SUBSCRIPTION,
+  STAKING_SUBSCRIPTION,
+} from './graphql';
 import styles from './Header.module.css';
 import { BlockHead, EraHead, SessionHead, StakingHead } from './types';
 
-const BLOCKS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeBlockNumbers {
-      authoredBy
-      hash
-      number
-      startDateTime
-    }
-  }
-`;
-
-const ERAS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeEras {
-      index
-      totalPoints
-    }
-  }
-`;
-
-const SESSIONS_SUBSCRIPTION = gql`
-  subscription {
-    subscribeSessions {
-      index
-    }
-  }
-`;
-
-const STAKING_SUBSCRIPTION = gql`
-  subscription {
-    subscribeStakes {
-      blockNumber {
-        number
-      }
-      totalStake
-    }
-  }
-`;
-
 const EraHeader = () => {
+  const { api } = useContext(ApiContext);
   const { data } = useSubscription(ERAS_SUBSCRIPTION);
+  const queryData = useQuery(LATEST_ERA_QUERY);
   const [eraHead, setEraHead] = useState<EraHead>();
 
   useEffect(() => {
     if (data) {
       const {
-        subscribeEras: { index, totalPoints },
+        subscribeEras: { index, individualPoints, totalPoints },
       } = data;
 
       if (!eraHead || index > eraHead.index) {
         setEraHead({
           index,
+          individualPoints,
           totalPoints,
         });
       }
     }
   }, [data, eraHead]);
 
+  useEffect(() => {
+    if (queryData && queryData.data) {
+      const {
+        data: { eras },
+      } = queryData;
+
+      setEraHead({
+        index: eras[0].index,
+        individualPoints: api.createType(
+          'Vec<Points>',
+          eras[0].individualPoints
+        ),
+        totalPoints: api.createType('Points', eras[0].totalPoints),
```

Comment:
I think we're fine for now. But it _might_ happen that this throws an error at some point, since we could be decoding past data with latest metadata/types.

Reply:
we could _maybe_ do something like you suggested for the other potential problem above + what we do in `node_watcher` such as:

"this is if my case for using the api above is valid"
```
export function createType(...args: AnyJSON) {
    // check spec version
    const runtimeVersion = await api.rpc.state.getRuntimeVersion(blockHash);
    const newSpecVersion = runtimeVersion.specVersion;

    // if spec version was bumped, update metadata in api registry
    if (newSpecVersion.gt(currentSpecVersion)) {
      l.warn(`bumped spec version to ${newSpecVersion}, fetching new metadata`);
      const rpcMeta = await api.rpc.state.getMetadata(blockHash);
      currentSpecVersion = newSpecVersion;

      // based on the node spec & chain, inject specific type overrides
      const chain = await api.rpc.system.chain();
      api.registry.register(getChainTypes(chain, runtimeVersion));

      api.registry.setMetadata(rpcMeta);
    }

    return api.createType(...args);
}
```

Pros:
* addresses the potential for a case where we'd need decode from an older version of metadata/types

Cons:
* it would mean 1-3 extra RPC calls per graphql subscription result (though this would be pretty infrequent, iff we show a different UI for when the DB has fallen behind the tip).

Reply:
What about never storing anything SCALE-encoded in the DB, so that on the frontend, we never need to do any `createType` (aka no SCALE-decoding on the front)?

Reply:
sounds good 👍(for the next minor version)